onpay.php

```diff
@@ -1340,4 +1370,55 @@ private function generateUrl($params) {
         $fullUrl = $baseUrl . '?' . http_build_query($params);
         return $fullUrl;
     }
+
+    private function getLatestModuleRelease() {
+        // If release file exists, examine it
+        if (file_exists(self::RELEASE_FILE)) {
+            $release = Release::fromString(file_get_contents(self::RELEASE_FILE));
+            // If release info is less than a day old, well simply return this info.
+            if (time() - $release->getLastCheck() < 86400) {
+                return $release;
+            }
+        }
+
+        $release = new Release();
+
+        // Fetch release info from OnPay
+        $releaseUri = 'https://api.onpay.io/plugin/release/prestashop';
+        $response = $this->httpGet($releaseUri);
+        // Examine response if it is present
+        if (null !== $response) {
+            // Set check timestamp
+            $release->setLastCheck(time());
```

Comment:
Consider caching negative results too, so if GitHub is in-accessible that is cached for 24 hours too, to avoid us causing hammering.